skglm/tests/test_prox_newton.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import numpy as np
+from numpy.linalg import norm
+from scipy.sparse import csc_matrix, issparse
+
+from sklearn.linear_model import LogisticRegression as LogReg_sklearn
+
+from skglm.penalties import L1, L1_plus_L2
+from skglm.datafits import Logistic
+from skglm.solvers.cd_solver import cd_solver
+from skglm.solvers.prox_newton_solver import prox_newton_solver
+
+from skglm.utils import make_correlated_data
+
+
+n_samples, n_features = 30, 50
```

Comment:
IMO: Define for every unitest its own params instead of defining them at the beginning.
Indeed,
- we will change the params for every unit test and thereby cover more cases
- when a unitest fails, the code will act as a standalone debug script

Reply:
It depends on what we want to test, and how we see the test script evolves. 
If we plan in later PRs to support other datafits, I suppose we are more interested in parametrizing the tests by the datafits/estimators rather than by the scale of the test setup. I adopted the same design as in https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/tests/test_estimators.py . 

No strong opinion on the matter though. @mathurinm WDYT?